sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -177,7 +184,15 @@ public ConfigurationAsyncClient buildAsyncClient() {
         policies.add(new RequestIdPolicy());
         policies.add(new AddHeadersPolicy(headers));
         policies.add(new AddDatePolicy());
-        policies.add(new ConfigurationCredentialsPolicy(buildCredential));
+
+        // add connection string credential if the AAD is not exist
+        if (tokenCredential == null) {
+            policies.add(new ConfigurationCredentialsPolicy(credential));
+        } else {
+            policies.add(
+                new BearerTokenAuthenticationPolicy(tokenCredential, String.format("%s/.default", buildEndpoint)));
+        }
```

Comment:
Need to validate that `tokenCredential` and `credential` aren't both null.

Reply:
ConfigurationCredentialsPolicy constructor checks credential is not null

Reply:
Correct but it would be confusing to the user as the stack trace for the exception, if it goes uncaught, will lead to either an implementation class or the `BearerTokenAuthenticationPolicy` which neither of them have been interacted with them directly. It would simplify and clarify by throwing it from the class they are interacting with.

Reply:
So it is better to remove the null check in ConfigurationCredentialsPolicy but check in the calling it method?

Reply:
Definitely leave the null check in ConfigurationCredentialsPolicy, just update this code to be something such as:

```java
if (credential != null) {
  // Use credential based policy
} else if (tokenCredential != null) {
  // User token based policy
} else {
  // Throw exception that credential and tokenCredential cannot be null
}
```

Reply:
So we prefer connection string over TokenCredential, if both of them exist?